File: app/src/main/java/igrek/songbook/chords/view/canvas/Align.java
package igrek.songbook.chords.view.canvas;

public class Align {
	public static final int DEFAULT = 0x000;
	//  Pozycja
	public static final int LEFT = 0x001;
	public static final int RIGHT = 0x002;
	public static final int HCENTER = 0x004;
	public static final int TOP = 0x010;
	public static final int BOTTOM = 0x020;
	public static final int VCENTER = 0x040;
	// mieszane
	public static final int CENTER = HCENTER | VCENTER;
	public static final int BOTTOM_LEFT = BOTTOM | LEFT;
	public static final int BOTTOM_RIGHT = BOTTOM | RIGHT;
	public static final int BOTTOM_HCENTER = BOTTOM | HCENTER;
	public static final int TOP_LEFT = TOP | LEFT;
	public static final int TOP_RIGHT = TOP | RIGHT;
	public static final int TOP_HCENTER = TOP | HCENTER;
	//  Rozmiar
	public static final int HADJUST = 0x100;
	public static final int VADJUST = 0x200;
	public static final int ADJUST = HADJUST | VADJUST;
}

File: app/src/main/java/igrek/songbook/songpreview/renderer/canvas/Align.kt
package igrek.songbook.songpreview.renderer.canvas

enum class Align(private val bits: Int) {

    LEFT(0x001),
    RIGHT(0x002),
    HCENTER(0x004),
    TOP(0x010),
    BOTTOM(0x020),
    VCENTER(0x040),

    CENTER(0x004 or 0x040),
    BOTTOM_LEFT(0x020 or 0x001),
    BOTTOM_RIGHT(0x020 or 0x002),
    BOTTOM_HCENTER(0x020 or 0x004),
    TOP_LEFT(0x010 or 0x001),
    TOP_RIGHT(0x010 or 0x002),
    TOP_HCENTER(0x010 or 0x004),
    ;

    fun isFlagSet(flag: Align): Boolean {
        return this.bits and flag.bits == flag.bits
    }
}